imports/plugins/included/taxes-taxcloud/client/containers/taxCloudSettingsFormContainer.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { composeWithTracker, registerComponent } from  "@reactioncommerce/reaction-components";
+import { Meteor } from "meteor/meteor";
+import { Packages } from "/lib/collections";
+import { Reaction, i18next } from "/client/api";
+import { TaxCloudPackageConfig } from "../../lib/collections/schemas";
+import { TaxCloudSettingsForm } from "../components";
+
+/**
+ * @file TaxCloudSettingsFormContainer is a React Component that serves
+ * as a container for TaxCloudSettingsForm.
+ * @module TaxCloudSettingsFormContainer
+ * @extends Component
+ */
+
+class TaxCloudSettingsFormContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  /**
+    * handleSubmit
+    * @method
+    * @summary event handler for when new TaxCloud settings are submitted.
+    * @param {Object} event - event info.
+    * @param {Object} changedInfo - info about the new TaxCloud settings.
+    * @param {String} targetField - where to save the new settings in the TaxCloud Package.
+    * @property {String} packageName - the name of this tax provider in lowercase.
+    * @since 1.5.2
+    * @return {null} - returns nothing
+    */
+  handleSubmit(event, changedInfo, targetField) {
+    if (!changedInfo.isValid) {
+      return;
+    }
+    Meteor.call("package/update", this.props.packageName, targetField, changedInfo.doc.settings.taxcloud, (error) => {
+      if (error) {
+        Alerts.toast(
+          i18next.t("admin.update.updateFailed", { defaultValue: "Failed to update TaxCloud settings." }),
+          "error"
+        );
+        return;
+      }
+      Alerts.toast(
+        i18next.t("admin.update.updateSucceeded", { defaultValue: "TaxCloud settings updated." }),
+        "success"
+      );
+    });
+  }
+
+  render() {
+    const providerName = this.props.providerName;
+    const packageSettings = this.props.settings;
+    const shownFields = {
+      [`settings.${providerName}.apiKey`]: TaxCloudPackageConfig._schema[`settings.${providerName}.apiKey`],
+      [`settings.${providerName}.apiLoginId`]: TaxCloudPackageConfig._schema[`settings.${providerName}.apiLoginId`]
+    };
+    const hiddenFields = [
+      `settings.${providerName}.enabled`,
+      `settings.${providerName}.refreshPeriod`,
+      `settings.${providerName}.taxCodeUrl`
+    ];
+
+    return (
```

Comment:
So this structure is still not right. Containers should not contain any presentation elements (no HTML). Just logic and data.

Reply:
Roger that. I'll change the structure.